CondFormats/RunInfo/interface/LHCInfoCombined.h

```diff
@@ -0,0 +1,55 @@
+#ifndef CondFormats_RunInfo_LHCInfoCombined_H
+#define CondFormats_RunInfo_LHCInfoCombined_H
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ESWatcher.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "CondFormats/RunInfo/interface/LHCInfo.h"
+#include "CondFormats/RunInfo/interface/LHCInfoPerLS.h"
+#include "CondFormats/RunInfo/interface/LHCInfoPerFill.h"
+
+#include "CondFormats/DataRecord/interface/LHCInfoPerLSRcd.h"
+#include "CondFormats/DataRecord/interface/LHCInfoPerFillRcd.h"
+#include "CondFormats/DataRecord/interface/LHCInfoRcd.h"
+
+#include "CondCore/CondDB/interface/Types.h"
+
+#include <bitset>
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+
+class LHCInfoCombined {
+public:
+  LHCInfoCombined() = default;
+
+  LHCInfoCombined(const LHCInfo& lhcInfo);
+  LHCInfoCombined(const LHCInfoPerLS& infoPerLS, const LHCInfoPerFill& infoPerFill);
+  LHCInfoCombined(const edm::EventSetup& iSetup,
```

Comment:
I think it would be better to use the other two constructors that take the input data products directly.

On the other hand, if this class is not really a data format class but a helper class (like the current use in `CTPPSProtonProducer` would suggest), it would be better placed in a non-dataformat package (in which case the `FWCore/Framework` dependence wouldn't be a problem).